connection.cpp

```diff
@@ -125,33 +145,56 @@ void UDPThread::stop() {
   fireTimer();
 }
 
+std::string UDPThread :: getAddressString( struct sockaddr_in * address ) {
+	char clientAddrStr[INET_ADDRSTRLEN];
+	std::string addr_string;
+
+	if (inet_ntop(AF_INET, &address->sin_addr, clientAddrStr, INET_ADDRSTRLEN) == NULL)	{
+		 lwarn << "Could not convert client address" << std::endl;
+	} 	else {
+	  addr_string.append(clientAddrStr);
+	}
+	return addr_string;
+}
+
+bool
+UDPThread::setClientConnectionTimeoutSec(uint32_t sec) {
+  bool retval = false;
+  if (m_bind2firstConnection) {
+    m_clientConnectionTimeoutSec = sec;
+    retval= true;
+  }
+  return retval;
+}
+
 void UDPThread::run() {
   fd_set readfds;
   ssize_t receivedBytes;
   uint8_t buffer[RECEIVE_BUFFER_SIZE];
-  struct sockaddr_in clientAddr;
   socklen_t clientAddrLen = sizeof(struct sockaddr_in);
-  char clientAddrStr[INET_ADDRSTRLEN];
 
   /* Set interval to m_timeout and an immediate timeout */
-  adjustTimer(m_timeout, 1);
+  adjustTimer(m_timerFdUdp, m_timeout, 1);
+
+  linfo << "UDPThread up and listenning on port: " << std::dec << m_localAddr.sin_port << std::endl;
+  linfo << "Remote Address is set to " << getAddressString(&m_remoteAddr) << std::endl;
 
-  linfo << "UDPThread up and running" << std::endl;
   while (m_started) {
     /* Prepare readfds */
     FD_ZERO(&readfds);
     FD_SET(m_udpSocket, &readfds);
-    FD_SET(m_timerfd, &readfds);
-    int ret = select(std::max(m_udpSocket,m_timerfd)+1, &readfds, NULL, NULL, NULL);
+    FD_SET(m_timerFdUdp, &readfds);
+    FD_SET(m_timerFdClientConnection, &readfds);
+    int ret = select(std::max(m_udpSocket,m_timerFdClientConnection)+1, &readfds, NULL, NULL, NULL);
```

Comment:
This should be `std::max({m_udpSocket,m_timerFdClientConnection,m_timerFdUdp})+1`